internal/service/credentials/resource_credential_type_test.go

```diff
@@ -0,0 +1,641 @@
+
+	p1Client, err := acctest.TestClient(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	apiClient := p1Client.API.CredentialsAPIClient
+	ctx = context.WithValue(ctx, credentials.ContextServerVariables, map[string]string{
+		"suffix": p1Client.API.Region.URLSuffix,
+	})
+
+	mgmtApiClient := p1Client.API.ManagementAPIClient
+	ctx = context.WithValue(ctx, management.ContextServerVariables, map[string]string{
+		"suffix": p1Client.API.Region.URLSuffix,
+	})
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "pingone_credential_type" {
+			continue
+		}
+
+		_, rEnv, err := mgmtApiClient.EnvironmentsApi.ReadOneEnvironment(ctx, rs.Primary.Attributes["environment_id"]).Execute()
+
+		if err != nil {
+
+			if rEnv == nil {
+				return fmt.Errorf("Response object does not exist and no error detected")
+			}
+
+			if rEnv.StatusCode == 404 {
+				continue
+			}
+
+			return err
+		}
+
+		body, r, err := apiClient.CredentialTypesApi.ReadOneCredentialType(ctx, rs.Primary.Attributes["environment_id"], rs.Primary.Attributes["id"]).Execute()
+
+		if err != nil {
+
+			if r == nil {
+				return fmt.Errorf("Response object does not exist and no error detected")
+			}
+
+			if r.StatusCode == 404 {
+				continue
+			}
+
+			tflog.Error(ctx, fmt.Sprintf("Error: %v", body))
+			return err
+		} else {
+
+			if body.DeletedAt != nil {
+
+				// Note: Credential Types are "soft delted" and may be returned via the ReadOneCredentialType call.
+				// If the DeletedAt attribute exists, it is considered deleted, handle similar to a 404.
+				return err
+			}
+
+		}
+
+		return fmt.Errorf("PingOne Credential Type ID %s still exists", rs.Primary.ID)
+	}
+
+	return nil
+}
+func TestAccCredentialType_Full(t *testing.T) {
+	t.Parallel()
+
+	resourceName := acctest.ResourceNameGen()
+	resourceFullName := fmt.Sprintf("pingone_credential_type.%s", resourceName)
+
+	name := acctest.ResourceNameGen()
+
+	data, _ := os.ReadFile("../../acctest/test_assets/image/credential_background_base64.png")
+	backgroundImage := "data:image/png;base64," + base64.StdEncoding.EncodeToString(data)
+
+	data, _ = os.ReadFile("../../acctest/test_assets/image/credential_logo_base64.png")
+	logoImage := "data:image/png;base64," + base64.StdEncoding.EncodeToString(data)
+
+	fullStep := resource.TestStep{
+		Config: testAccCredentialTypeConfig_Full(resourceName, name, backgroundImage, logoImage),
+		Check: resource.ComposeTestCheckFunc(
+			resource.TestMatchResourceAttr(resourceFullName, "id", verify.P1ResourceIDRegexp),
+			resource.TestMatchResourceAttr(resourceFullName, "environment_id", verify.P1ResourceIDRegexp),
+			resource.TestCheckResourceAttr(resourceFullName, "title", name),
+			resource.TestCheckResourceAttr(resourceFullName, "description", fmt.Sprintf("%s Example Description", name)),
+			resource.TestCheckResourceAttr(resourceFullName, "card_type", "VerifiedEmployee"),
+			resource.TestCheckResourceAttrSet(resourceFullName, "card_design_template"),
```

Comment:
We will need to validate the full `card_design_template` string ideally

Reply:
Added additional card design template tests that align to the validation rules.  Future consideration is to introduce a full XML parser, and strengthen the approach to interrogate the template.

Reply:
added validation of full string also